### DISCUSSION TITLE
Has the red service really the need to store the orderId with the rateId selected?

### DISCUSSION THREAD

**Discussion by antonio8101** [artiso-solutions/adsd-hotel#39] [General] [answered]
We plan to have a command InputRatesSelected (maybe better SelectRates), where We are supposed to do something and emit and event InputRateSelected.
We found that can probably be not important for the red service to store the OrderId, since the request information really rely on our RateId?
When a rate has been selected the red service simply can emit an event containing the RateId (and some other important information on the rate) with the OrderId provided by its ambassador, without caring to store something in its storage.
Then if, after the booking, someone ask to the red service the rate id selected, they can simply rely on our RateId.
And also can be a little redundant having that information stored in our service beside other services that do the same on purpose (maybe it's the case of the yellow service, or some other in the system)..

What do you think about it?

**Comment by tommasobertoni** [comment, marked as answer]
During the Check In, we want to display all the data relative to a booking.
Can you briefly describe which services, API calls and data would be needed to achieve this, within your proposal?

**Reply by exbachi** [reply]
If we want to show the all the informations to an order, there is no endpoint specified on the red service yet. 
So we should safe the data to be able to extend the ambassador interface in a later step?

**Reply by tziemek** [reply]
Yes, basically every service need to provide an api to retrieve the data via OrderId.
